File: src/main/java/me/beanbag/utils/LitematicaUtils.java
package me.beanbag.utils;

import me.beanbag.Nuker;
import me.beanbag.utils.litematica.LitematicaHelper;
import net.minecraft.util.math.BlockPos;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

public class LitematicaUtils {
    public static Set<BlockPos> schematicMismatches = Collections.synchronizedSet(new HashSet<>());
    public static void updateLitematicaBlocks() {
        // Litematica
        if (Nuker.litematica) {
            if (!LitematicaHelper.isLitematicaLoaded()) {
                Nuker.litematica = false;
            } else {
                schematicMismatches = LitematicaHelper.INSTANCE.getMismatches();
            }
        }
    }
}

File: src/main/kotlin/me/beanbag/nuker/utils/LitematicaUtils.kt
package me.beanbag.nuker.utils

import fi.dy.masa.litematica.data.DataManager.getSchematicPlacementManager
import me.beanbag.nuker.mixins.litematica.ISchematicVerifierAccessor
import net.minecraft.util.math.BlockPos

object LitematicaUtils {
    var schematicIncorrectBlockPlacements = hashSetOf<BlockPos>()
    var schematicIncorrectStatePlacements = hashSetOf<BlockPos>()

    fun updateSchematicMismatches() {
        schematicIncorrectBlockPlacements.clear()
        schematicIncorrectStatePlacements.clear()

        for (placement in getSchematicPlacementManager().allSchematicsPlacements) {
            if (!placement.isEnabled) continue
            val schematicVerifier = placement.schematicVerifier as ISchematicVerifierAccessor
            schematicIncorrectBlockPlacements.addAll(schematicVerifier.wrongBlocksPositions.values())
            schematicIncorrectStatePlacements.addAll(schematicVerifier.wrongStatesPositions.values())
        }
    }
}